airflow/models/taskinstance.py

```diff
@@ -1410,18 +1410,7 @@ def _execute_task_with_callbacks(self, context, test_mode=False):
         """Prepare Task for Execution"""
         from airflow.models.renderedtifields import RenderedTaskInstanceFields
 
-        parent_pid = os.getpid()
```

Comment:
That's why it needs to be separated to a separate PR where you explain your intentions in detail in the PR description. This informationw was missing in the original commit "Remove dag parsing in airflow run" did not explain any of the context here.

Reply:
BTW. I think we should not remove forking. If you want to remove it , then this should be a separate PR where you explain why. Forking is shaving multiple seconds when startting the task sometimes because it does not have to launch the interpreter. I'd even say that we won't remove forking because of that. There shuld be very good reason why we would like to remove it and some justification on how it impacts the task start time.

Reply:
@potiuk actually on a second thoughts, these 2 commits are intertwined. How about let me merge those 2 commits into single one and add more description to the commit message and the PR description. 

let me know your thoughts.